File: jobs/2026-05-27__01-46-48/activity_get_request_ts__Nkabdus/artifacts/user/myproject/src/activities.ts
/**
 * activities.ts
 *
 * Activities are the correct place for all I/O (network calls, file system, etc.)
 * because they run in a normal Node.js context and are retried by Temporal on failure.
 * The workflow code itself must remain deterministic and must never perform I/O directly.
 */

/**
 * Performs a real HTTP GET request to the given URL and returns the response body
 * as a plain string. Uses the global `fetch` available in Node 18+.
 */
export async function fetchData(url: string): Promise<string> {
  const response = await fetch(url, {
    headers: {
      // GitHub API requires a User-Agent header
      'User-Agent': 'temporal-fetch-url-workflow/1.0',
    },
  });

  if (!response.ok) {
    throw new Error(
      `HTTP request failed: ${response.status} ${response.statusText} for URL ${url}`
    );
  }

  return await response.text();
}

File: jobs/2026-05-27__01-46-48/activity_get_request_py__XUwYUDi/artifacts/user/myproject/activities.py
import httpx
from temporalio import activity

_HEADERS = {"User-Agent": "temporal-fetch-workflow/1.0"}


@activity.defn
async def fetch_data(url: str) -> str:
    """Perform a real HTTP GET and return the response body as text."""
    async with httpx.AsyncClient(headers=_HEADERS, follow_redirects=True) as client:
        response = await client.get(url)
        # Return body for all responses; the workflow gets the text regardless
        # of HTTP status so that transient upstream errors surface as a result
        # rather than an unhandled exception that causes the workflow to fail.
        return response.text
